src/audio/audio_system.c

```diff
@@ -67,7 +71,13 @@ static u32 AudioEventSegment_GetCount(const AudioEventSegment* seg);
 static bool AudioEventSegment_Push(AudioEventSegment* seg, const AudioEvent* evtIn);
 static bool AudioEventSegment_Pop(AudioEventSegment* seg, AudioEvent* evtOut);
 static void OnMidiEventFn(const MidiMsg* msg, void* usr);
-static void OnAudioPacketFn(float* buffer, i32 ticks, i32 numChannels);
+static int OnAudioPacketFn(
+    const void* inputBuffer,
+    void* outputBuffer,
+    unsigned long frameCount,
```

Comment:
I'm unsure how best to handle this particular case of implementing a callback -- If I use `u64` here I get a warning on Windows, because `u64` is not defined as `unsigned long` on windows. Do we just stick with `unsigned long` to stay consistent with the PortAudio API?